app/shared/api-services/poll/poll-factory.js

```diff
@@ -11,71 +11,63 @@ angular.module('apiServices.poll.factory', [
       filter,
       pollUrl = '/cms/api/v1/poll/';
 
+  var error = function(message) {
+    return new Error('Poll Error: ' + message);
+  };
+
   function getPoll(pollId) {
     filter = $filter('date_string_to_moment');
 
     return $http.get(pollUrl + pollId)
     .then(function (response) {
-      if(response.status === 200) {
-        response.data.end_date = filter(response.data.end_date);
-        return response.data;
-      } else {
-        return $q.reject('Unable to retrieve poll');
-      }
+      response.data.end_date = filter(response.data.end_date);
+      return response.data;
     });
   }
 
   function postPoll(data) {
-    filter = $filter('moment_to_date_string');
-
-    pollInfo = {
-      title: data.title,
-      question_text: data.question_text
-    };
+    if(!data.title && !data.question_text) {
+      throw error('title and question text required');
+    }
 
     if(data.end_date) {
+      if(typeof data.end_date !== 'object') {
```

Comment:
This check should exist inside the `moment_to_date_string` filter. Also, use the [`moment.isMoment` method](http://momentjs.com/docs/#/query/is-a-moment/).